src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -88,8 +88,8 @@ async function kdf(hash_algo, X, length, param, stripLeading=false, stripTrailin
 async function genPublicEphemeralKey(curve, Q) {
   switch (curve.name) {
     case 'curve25519': {
-      const { secretKey: d } = nacl.box.keyPair();
-      const { secretKey, sharedKey } = await genPrivateEphemeralKey(curve, Q, d);
+      const { publicKey: recipientPublicKey, secretKey: d } = nacl.box.keyPair();
+      const { secretKey, sharedKey } = await genPrivateEphemeralKey(curve, Q, d, recipientPublicKey);
       let { publicKey } = nacl.box.keyPair.fromSecretKey(secretKey);
```

Comment:
I know that `recipientPublicKey` is not used in this case, but I think it is not correct. The reason is that `genPrivateEphemeralKey` does some processing on the private key before using it. The `publicKey` here is the real one. Maybe you could just pass `null` to `genPrivateEphemeralKey` for now, in order to avoid confusion?
(Maybe we can reorganize this code in a later PR to calculate the public key first, so that it makes more sense.)

Reply:
resolved